tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -15,38 +16,63 @@
 
 
 class TestModelExport(unittest.TestCase):
-    def setUp(self):
-        iris = datasets.load_iris()
-        self._X = iris.data[:, :2]  # we only take the first two features.
-        self._y = iris.target
-        self._trainingFeatures = {}
-        self._feature_names = iris.feature_names[:2]
-        for i in range(0, 2):
-            # Tensorflow is fickle about feature names, so we remove offending characters
-            iris.feature_names[i] = iris.feature_names[i].replace(" ", "")
-            iris.feature_names[i] = iris.feature_names[i].replace("(", "")
-            iris.feature_names[i] = iris.feature_names[i].replace(")", "")
-            self._trainingFeatures[iris.feature_names[i]] = iris.data[:, i:i+1]
-        self._tf_feat_cols = []
-        self._feature_names = iris.feature_names[:2]
-        # Creating Tensorflow-specific numeric columns for input.
-        for col in iris.feature_names[:2]:
-            self._tf_feat_cols.append(tf.feature_column.numeric_column(col))
-        # Creating input training function.
-        self._input_train = tf.estimator.inputs.numpy_input_fn(self._trainingFeatures, 
-                                                                    self._y, 
-                                                                    shuffle=False, 
-                                                                    batch_size=1)
-        # Creating Deep Neural Network Regressor. 
-        self._dnn = tf.estimator.DNNRegressor(feature_columns=self._tf_feat_cols, 
-                                                hidden_units=[1])
-        self._sess = tf.Session()
-        # Training and creating expected predictions on training dataset.
-        self._dnn.train(self._input_train, steps=100)
-        self._dnn_predict = self._dnn.predict(self._input_train)
+
+    def helper(self, feature_spec, tmp, model, model_predict, df):
+        # This functions handles 
+        receiver_fn = tf.estimator.export.build_raw_serving_input_receiver_fn(feature_spec)
+        saved_model_path = tmp.path("model")
+        os.makedirs(saved_model_path)
+        os.makedirs(tmp.path("hello"))
+        # Saving Tensorflow model.
+        saved_model_path = model.export_savedmodel(saved_model_path, 
+                                                       receiver_fn).decode("utf-8")
+        # Logging the Tensorflow model just saved.
+        tensorflow.log_saved_model(saved_model_dir=saved_model_path,
+                                   signature_def_key="predict", 
+                                   artifact_path=tmp.path("hello"))
+        # Loading the saved Tensorflow model as a pyfunc.
+        x = pyfunc.load_pyfunc(saved_model_path)
+        # Predicting on the dataset using the pyfunc.
+        xpred = x.predict(df)
+        saved = []
+        for s in model_predict:
```

Comment:
As per the comment lower down, I think this method can just `return [row for _, row in xpred.iterrows()]` (returning only the pyfunc predictions since we already know the `saved` estimator predictions, using a list-comprehension for conciseness)